### DISCUSSION TITLE
Performance issues with the Osu!Lazer APK version.

### DISCUSSION THREAD

**Discussion by BlackPistol** [ppy/osu#15686] [Q&A] [answered]
Issues with Osu!Lazer so far:
- Performance issues when viewing display (Images of the beatmaps, ranks, etc..)
- Performance issues when playing any beatmaps of any difficulty.
- Taiko, Catch and Mania gamemodes aren't supported on the APK version of Osu!Lazer. Meaning that they aren't playable.
- Harder difficulty beatmaps have bad tempo (timing).
- Osu!Lazer should have a storage location for the beatmaps, skins, replays and scores.

**Comment by peppy** [comment, marked as answer]
Great

**Reply by peppy** [reply]
Performance and optimisation considerations are an ongoing effort. Especially on android devices, where the hardware can be of varying quality, it may be impossible to run well on some devices. Please be patient and wait for future optimisations, or upgrade your device.

**Reply by BlackPistol** [reply]
Will the project be completed someday?

**Reply by peppy** [reply]
No. osu! has been evolving for over a decade and will continue to do so for the foreseeable future.

**Reply by BlackPistol** [reply]
What about the other three gamemodes? Like Taiko, Catch and Mania?

**Reply by peppy** [reply]
same applies
